### PR TITLE
[LW-10649] Fix installing & running with strict AppArmor

### PR DESCRIPTION
**Pull request by michalrus**
[LW-10649](https://input-output.atlassian.net/browse/LW-10649)

… by removing the `nix-chroot`.

The issue manifested on Ubuntu 24.04 LTS

## Todos

_n/a_

## Screenshots

_n/a_

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [ ] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

[LW-10649]: https://input-output.atlassian.net/browse/LW-10649?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: nix/internal/x86_64-linux.nix [hunk 1]
```diff
@@ -170,158 +170,320 @@ in rec {
...
+    name = "Daedalus-${cluster}";
+    exec = "INSERT_PATH_HERE";
+    desktopName = "Daedalus ${cluster}";
+    genericName = "Crypto-Currency Wallet";
+    categories = [ "Application" "Network" ];
+    icon = "INSERT_ICON_PATH_HERE";
+  });
 
-    preInstall = pkgs.writeText "pre-install" ''
-      if grep sse4 /proc/cpuinfo -q; then
```
Comment (michalrus) [inline nix/internal/x86_64-linux.nix:292]: @cleverca22 says this needs to be added back 📝

File: source/main/ipc/manageAppUpdateChannel.ts [hunk 2]
```diff
@@ -169,7 +168,11 @@ export const handleManageAppUpdateRequests = (window: BrowserWindow) => {
           return reject();
         }
 
-        safeExitWithCode(20);
+        // We need to also wait for `cardano-node` to exit cleanly, otherwise the new auto-updated version
+        // is showing the new node crashing, because the old one is still running for around 30 seconds:
+        // WRONG: safeExitWithCode(20);
+        (window as any).daedalusExitCode = 20;
```
[the remark this answers is not in the document]
Comment (danielmain) [reply source/main/ipc/manageAppUpdateChannel.ts:174]: Will safeExitWithCode be executed later? what happened if we do not set the log level to false (which happened in the safeExitWithCode)

File: source/main/utils/buildAppMenus.ts [hunk 3]
```diff
@@ -64,7 +63,10 @@ export const buildAppMenus = async (
     logger.info('Exiting Daedalus with code 21', {
       code: 21,
     });
-    safeExitWithCode(21);
+    // We have to make sure that cardano-node exits, otherwise we get DB locked errors at startup:
+    // WRONG: safeExitWithCode(21);
```
Comment (danielmain) [inline source/main/utils/buildAppMenus.ts:67]: Looks correct, safeExit is triggered when the Window is closed via `safeExit`, so no need to comment out the old code with `safeExitWithCode`

**Review by danielmain — APPROVED** [review]
You do not need to comment out the old safeExitWithCode, I see you handle exits correctly.

File: source/main/ipc/manageAppUpdateChannel.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (michalrus) [reply source/main/ipc/manageAppUpdateChannel.ts:174]: @danielmain says it's fine :)

File: nix/internal/x86_64-linux.nix (hunk 1 above)
Comment (michalrus) [reply nix/internal/x86_64-linux.nix:292]: Added back in b1441dd8b12ddcf2d4491569297f4c5c7d1ed87b

File: source/main/utils/buildAppMenus.ts (hunk 3 above)
Comment (michalrus) [reply source/main/utils/buildAppMenus.ts]: Fixed in ab60ea2c8db169fa2c4e1fa149d6592059b80765

**Pull request merged**